pkg/client/neutron.go

```diff
@@ -37,115 +35,162 @@ func newNeutronV2(providerClient *gophercloud.ProviderClient, eo gophercloud.End
+	opts := networks.ListOpts{
+		Name: name,
+	}
+
+	allPages, err := networks.List(n.serviceClient, opts).AllPages(ctx)
+	onCall("neutron")
+	if err != nil {
+		onFailure("neutron")
+		return "", fmt.Errorf("failed to list networks: %w", err)
+	}
 
-	metrics.APIRequestCount.With(prometheus.Labels{"provider": "openstack", "service": "neutron"}).Inc()
+	allNetworks, err := networks.ExtractNetworks(allPages)
 	if err != nil {
-		metrics.APIFailedRequestCount.With(prometheus.Labels{"provider": "openstack", "service": "neutron"}).Inc()
-		return "", err
+		onFailure("neutron")
+		return "", fmt.Errorf("failed to extract networks: %w", err)
+	}
+
+	for _, net := range allNetworks {
+		if net.Name == name {
+			return net.ID, nil
+		}
 	}
-	return id, nil
+
+	return "", fmt.Errorf("no network found with name: %s", name)
 }
 
 // GroupIDFromName resolves the given security group name to a unique ID.
-func (n *neutronV2) GroupIDFromName(name string) (string, error) {
-	id, err := utilGroups.IDFromName(n.serviceClient, name)
+func (n *neutronV2) GroupIDFromName(ctx context.Context, name string) (string, error) {
+	listOpts := groups.ListOpts{
+		Name: name,
+	}
+
+	allPages, err := groups.List(n.serviceClient, listOpts).AllPages(ctx)
+	onCall("neutron")
+	if err != nil {
+		onFailure("neutron")
+		return "", fmt.Errorf("failed to list security groups: %w", err)
+	}
 
-	metrics.APIRequestCount.With(prometheus.Labels{"provider": "openstack", "service": "neutron"}).Inc()
+	allGroups, err := groups.ExtractGroups(allPages)
 	if err != nil {
-		metrics.APIFailedRequestCount.With(prometheus.Labels{"provider": "openstack", "service": "neutron"}).Inc()
-		return "", err
+		return "", fmt.Errorf("failed to extract security groups: %w", err)
 	}
-	return id, nil
+
+	for _, group := range allGroups {
+		if group.Name == name {
+			return group.ID, nil
+		}
+	}
+
+	return "", fmt.Errorf("no security group found with name: %s", name)
 }
 
 // PortIDFromName resolves the given port name to a unique ID.
-func (n *neutronV2) PortIDFromName(name string) (string, error) {
-	id, err := utilPorts.IDFromName(n.serviceClient, name)
-	metrics.APIRequestCount.With(prometheus.Labels{"provider": "openstack", "service": "neutron"}).Inc()
+func (n *neutronV2) PortIDFromName(ctx context.Context, name string) (string, error) {
+	opts := ports.ListOpts{
+		Name: name,
+	}
+
+	allPages, err := ports.List(n.serviceClient, opts).AllPages(ctx)
+	onCall("neutron")
+	if err != nil {
+		onFailure("neutron")
+		return "", fmt.Errorf("failed to list ports: %w", err)
+	}
 
+	allPorts, err := ports.ExtractPorts(allPages)
 	if err != nil {
-		metrics.APIFailedRequestCount.With(prometheus.Labels{"provider": "openstack", "service": "neutron"}).Inc()
-		return "", err
+		onFailure("neutron")
+		return "", fmt.Errorf("failed to extract ports: %w", err)
+	}
+
+	for _, port := range allPorts {
+		if port.Name == name {
+			return port.ID, nil
```

Comment:
this may not work as expected when there are multiple ports with the same name. I believe the old implementation returned an error in that case but I am not sure. Likewise for all similar functions